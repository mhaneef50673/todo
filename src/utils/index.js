import { LOCAL_STORAGE_KEY } from '../constants';

export const isAuthenticated = () => {
  const token = localStorage.getItem(LOCAL_STORAGE_KEY);
  if (token) {
    return true;
  }
  return false;
}