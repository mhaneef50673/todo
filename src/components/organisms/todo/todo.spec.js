import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import { Provider } from 'react-redux';
import Todo from './index';
import { isAuthenticated } from '../../../utils';

jest.mock('../../../utils', () => ({
  isAuthenticated: jest.fn(),
}));

describe('Todo Component', () => {
  beforeEach(() => {
    isAuthenticated.mockImplementation(() => true);
  });

  it('should render the component', () => {
    const tree = mount(
      <Provider store={{
        getState: () => (
          {
            loginReducer: {
              token: 'test',
              error: false,
              isAuthenticating: false,
            },
            todoReducer: {
              todos: [
                {
                  description: 'item 1',
                  status: 'pending',
                },
              ]
            }
          }
        ),
        subscribe: () => { },
        dispatch: () => { },
      }}
      >
        <Todo />
      </Provider>
    )
    expect(mountToJson(tree)).toMatchSnapshot();
  });
});