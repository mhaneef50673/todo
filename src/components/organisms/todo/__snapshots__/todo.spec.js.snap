// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Component should render the component 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Todo)>
    <Todo
      addTodoItem={[Function]}
      logout={[Function]}
      todos={
        Array [
          Object {
            "description": "item 1",
            "status": "pending",
          },
        ]
      }
      toggleTodoStatus={[Function]}
      token="test"
    >
      <Button
        action={[Function]}
        buttonText="logout"
        clickHandler={[Function]}
        extraClass="primary"
      >
        <button
          className="primary"
          onClick={[Function]}
          type="button"
        >
          logout
        </button>
      </Button>
      <TodoView
        addTodoItem={[Function]}
        list={
          Array [
            Object {
              "description": "item 1",
              "status": "pending",
            },
          ]
        }
        onTodoItemClickHandler={[Function]}
      >
        <h1>
           Todo List 
        </h1>
        <div
          className="todo-view"
        >
          <Memo(TodoItem)
            index={0}
            key="0"
            onTodoItemClickHandler={[Function]}
            status="pending"
          >
            <div
              className="todo-item pending"
              onClick={[Function]}
            >
              <span
                className="text"
              />
              <span
                className="text"
              >
                pending
              </span>
            </div>
          </Memo(TodoItem)>
        </div>
        <div
          className="add-item"
        >
          <div
            className="center"
          >
            <Button
              action={[Function]}
              buttonText="Add Item"
              clickHandler={[Function]}
              extraClass="primary"
            >
              <button
                className="primary"
                onClick={[Function]}
                type="button"
              >
                Add Item
              </button>
            </Button>
          </div>
        </div>
        <div
          className="modal-container"
        >
          <Modal
            isVisible={false}
            title="Enter item Details"
          />
        </div>
      </TodoView>
    </Todo>
  </Connect(Todo)>
</Provider>
`;
